src/wallet/wallet.ts

```diff
@@ -1707,34 +1722,20 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
         throw new SendTxError('The melt authority address must belong to your wallet.');
       }
     }
+    const tokenInfoVersion = newOptions.tokenInfoVersion || TokenVersion.DEPOSIT;
 
-    const depositPercent = this.storage.getTokenDepositPercentage();
     // 1. Calculate HTR deposit needed
-    let deposit = tokens.getDepositAmount(amount, depositPercent);
+    let deposit = 0n;
+    if (tokenInfoVersion === TokenVersion.DEPOSIT) {
+      const depositPercent = this.storage.getTokenDepositPercentage();
+      deposit += tokens.getDepositAmount(amount, depositPercent);
+    }
 
     if (newOptions.data && newOptions.data.length > 0) {
-      // For data outputs, we have a fee of 0.01 HTR per data output
+      // For data outputs, we have a fee of 0.01 HTR per data output (this fee isn't related to the transaction fee that is calculated based in the token version)
       deposit += tokens.getDataFee(newOptions.data.length);
     }
 
-    // 2. Get utxos for HTR
```

Comment:
There seems to be a numbering for each step being made and it was broken when you moved this step, maybe we should remake these comments?